### PR TITLE
Refactor webpack configuration

### PR DESCRIPTION
**Pull request by erikyuzwa**
- updated `concurrently` and `electron` packages
- added `dotenv` package (and support)
- refactor webpack configurations
- use `webpack-merge` to combine the common configuration with any extra configuration
- updated README for pixi support
- update script convention
```
npm run build:electron
npm run build:electron:pixi

npm run web
npm run web:pixi

npm run package:win32
npm run package:win32:pixi

```

### CODE REVIEW

File: src/index.html [hunk 1]
```diff
@@ -4,16 +4,19 @@
...
 	<meta name="viewport" content="width=device-width, initial-scale=1">
-	<% if (process.env.NODE_ENV==="production" ) { %>
-		<meta http-equiv="Content-Security-Policy" content="script-src 'self' 'unsafe-inline' 'unsafe-eval'; style-src 'self' 'unsafe-inline' https://fonts.googleapis.com/;  img-src 'self' ">
+	<% if (process.env.NODE_ENV === "production" ) { %>
+		<meta http-equiv="Content-Security-Policy" content="script-src 'self' 'unsafe-inline' 'unsafe-eval' blob; style-src 'self' 'unsafe-inline' https://fonts.googleapis.com/;  img-src 'self';worker-src 'self' blob: ">
 	<% } %>
-	<title>JSRL</title>
+	<title><%= htmlWebpackPlugin.options.title %></title>
 	<link href="./unicodetiles/unicodetiles.css" rel="stylesheet" type="text/css" />
 	<script src="./unicodetiles/unicodetiles.js"></script>
```
Comment (slashman) [inline src/index.html:12]: I'm going to refactor stuff so we are not tied to this dependency for pixi deployments, but we can fix that later.

File: src/index.html [hunk 2]
```diff
@@ -4,16 +4,19 @@
...
 	<script src="./unicodetiles/input.js"></script>
```
Comment (slashman) [inline src/index.html:13]: We'll also eventually use a separate lib or just vanilla JS for input, removing this dependency.

**Comment by slashman** [comment]
looks good overall! only thing I'd change is the display type name 'default' for the unicode tiles visuals. I'm thinking it should be 'unicodetiles' instead since they should be on equal level (I even think, in the future, the default visuals will be pixi probably).

In any case, not a blocker for merging; will don once I verify it works well.

**Comment by erikyuzwa** [comment]
I like that idea - renamed!

I did notice that `target` still doesn't quite come through "cleanly" due to `electron-renderer` which I identified as a thing to deprecate. I'm thinking the default `target` should be `web`, then we apply electron when required

File: src/index.html (hunk 1 above)
Comment (erikyuzwa) [reply src/index.html:12]: cool

File: src/index.html (hunk 2 above)
Comment (erikyuzwa) [reply src/index.html:13]: k sounds good

**Comment by erikyuzwa** [comment]
I've got ONE more very very minor webpack config cleanup to pursue (not renaming anything), and then I'll stop to look at other things to work on here...code, etc.

**Review by slashman — APPROVED** [review]
Good progress, there's an outstanding TODO with a proper issue for it (https://github.com/slashman/jsrl/issues/9)

**Comment by slashman** [comment]
Note that there are still issues with concurrency for the `npm run electron:*` tasks.

**Comment by erikyuzwa** [comment]
Agreed - created https://github.com/slashman/jsrl/issues/13

**Pull request merged**